app/src/main/java/com/twilio/voice/quickstart/IncomingCallNotificationService.java

```diff
@@ -151,7 +153,10 @@ private void handleCancelledCall(Intent intent) {
     }
 
     private void handleIncomingCall(CallInvite callInvite, int notificationId) {
-        startForeground(notificationId, createNotification(callInvite, notificationId));
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O ||
```

Comment:
Let's research using the [DefaultLifecycleObserver](https://developer.android.com/reference/androidx/lifecycle/DefaultLifecycleObserver.html) to determine if the app is visible. Change the channel priority based on the app visibility.

- App is visible. Just a notification with actions. Not heads up
- App is not visible. Heads up notification with actions.